src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final float LOAD_fACTOR = 0.75f;
+    private Node<K, V>[] table = new Node[DEFAULT_CAPACITY];
+    private int threshold = (int) (DEFAULT_CAPACITY * LOAD_fACTOR);
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        Node<K, V> node = new Node<>(getHash(key), key, value, null);
+        int index = getBucket(key, table.length);
+        if (table[index] == null) {
+            table[index] = node;
+            size++;
+        } else {
+            for (Node<K, V> var = table[index]; var != null; var = var.next) {
+                if (key == var.key
+                        || getHash(key) == var.hash && key != null && key.equals(var.key)) {
+                    var.value = value;
+                    break;
+                }
+                if (var.next == null) {
+                    var.next = node;
+                    size++;
+                    break;
+                }
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getBucket(key, table.length);
+        for (Node<K, V> var = table[index]; var != null; var = var.next) {
+            if (key == var.key
+                    || getHash(key) == var.hash && key != null && key.equals(var.key)) {
+                return var.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        if (size == threshold) {
+            Node<K, V>[] tempTable = table;
+            table = new Node[table.length * 2];
+            for (Node<K, V> nodeBucket : tempTable) {
+                if (nodeBucket != null) {
+                    for (Node<K, V> temp = nodeBucket; temp != null; temp = temp.next) {
+                        int index = getBucket(temp.key, table.length);
```

Comment:
you don't need to calculate index. Before this loop execute `size = 0`; and here just iterate each node and call `add(key, value);`